pyshtools/shclasses/shcoeffsgrid.py

```diff
@@ -629,6 +633,91 @@ def from_cap(self, theta, lmax, clat=None, clon=None, normalization='4pi',
                                    degrees=False)
 
         return temp
+    
+    @classmethod
+    def from_netcdf(self, filename, lmax=None, normalization='4pi', csphase=1):
+        """
+        Initialize the class with spherical harmonic coefficients from a 
+        netcdf file.
+
+        Usage
+        -----
+        x = SHCoeffs.from_netcdf(filename)
+
+        Returns
+        -------
+        x : SHCoeffs class instance.
+        
+        Parameters
+        ----------
+        filename : str
+            Name of the file, including path.
+        lmax : int, optional, default = None
+            The maximum spherical harmonic degree to read.
+        normalization : str, optional, default = '4pi'
+            '4pi', 'ortho', 'schmidt', or 'unnorm' for geodesy 4pi normalized,
+            orthonormalized, Schmidt semi-normalized, or unnormalized
+            coefficients, respectively.
+        csphase : int, optional, default = 1
+            Condon-Shortley phase convention: 1 to exclude the phase factor,
+            or -1 to include it.
+            
+        Description
+        -----------
+        The format of the netcdf file has to be exactly as the format that is
+        used in SHCoeffs.to_netcdf().
+        """
+        if type(normalization) != str:
+            raise ValueError('normalization must be a string. '
+                             'Input type was {:s}'
+                             .format(str(type(normalization))))
+
+        if normalization.lower() not in ('4pi', 'ortho', 'schmidt', 'unnorm'):
+            raise ValueError(
+                "The input normalization must be '4pi', 'ortho', 'schmidt', "
+                "or 'unnorm'. Provided value was {:s}"
+                .format(repr(normalization))
+                )
+
+        if csphase != 1 and csphase != -1:
+            raise ValueError(
+                "csphase must be 1 or -1. Input value was {:s}"
+                .format(repr(csphase))
+                )
+    
+        if not filename[-3:] == '.nc':
```

Comment:
I think that this is a good idea if `filename` does not have an extension. However, if you provide an extension like `filename.netcdf` then I don't think we need to append another extension `.nc`.

Reply:
Alright, I deleted the lines. Generally I also think it is a good idea, but it would be inconsistent with the other functions anyways.